lib/checkmemoryleak.cpp

```diff
@@ -488,16 +500,35 @@ void CheckMemoryLeakInFunction::checkReallocUsage()
 
         // Search for the "var = realloc(var, 100" pattern within this function
         for (const Token *tok = scope->bodyStart->next(); tok != scope->bodyEnd; tok = tok->next()) {
-            if (tok->varId() > 0 &&
-                Token::Match(tok, "%name% = realloc|g_try_realloc ( %name% ,") &&
-                tok->varId() == tok->tokAt(4)->varId() &&
-                isNoArgument(symbolDatabase, tok->varId())) {
+            if (tok->varId() > 0 && Token::Match(tok, "%name% =")) {
+                const Token* reallocTok = tok->tokAt(2);
+                if (reallocTok && reallocTok->str() == "(") {
```

Comment:
This code looks fragile.. I think `Token::isCast` method can be used here to make the code more robust. Please consider such code:

    p = (realloc(p,100));

.. it's probably better to use the AST to look at assignment RHS.

Reply:
I tried that, but didn't get it to work so I more or less copied how casts were treated in other places in the file, but I now realize that it's because the checks are run on the full token list, but the tests are still run on the simplified token list. I think this also explains why I never managed to write a test case that failed when removing these casts, but I needed them there to get `make checkcfg` working...

I'll update it.

Reply:
Hmm, as far as I can see, no checks run on the simplified token list anymore, but quite a few testcases run `simplifyTokenList2()`. This must be incorrect?

Reply:
> Hmm, as far as I can see, no checks run on the simplified token list anymore, but quite a few testcases run simplifyTokenList2(). This must be incorrect?

yes that is very unfortunate.